maps/ascii.py

```diff
@@ -35,27 +36,32 @@ class AsciiSphericalMap(AsciiMap, Map):
     A map represented as ASCII characters.
     """
 
-    def get_frame_at(self, x_start: int, y_start: int, x_end: int, y_end: int) -> Frame:
+    def get_frame_at(self, frame_coords: FrameCoords) -> Frame:
         """
         Allows retrieval of frames assuming the map is spherical and coordinates
         can wrap (i.e. x_end and y_end coordinates can be smaller than x_start and
         y_start).
 
-        :param x_start: X coordinate of top-left corner of the frame.
-        :param y_start: Y coordinate of top-left corner of the frame.
-        :param x_end: X coordinate of bottom-right corner of the frame.
-        :param y_end: Y coordinate of bottom-right corner of the frame.
+        :param frame_coords: The coordinates of the frame
         :return: The frame.
         """
         frame = Frame([])
-        if y_start <= y_end:
-            for row in self.representation[y_start:y_end + 1]:
-                frame.append(self.get_row_subset(row, x_start, x_end))
+        if frame_coords.y_top <= frame_coords.y_bottom:
```

Comment:
One side benefit of having a `FrameCoords` instead of a list of lists is that the code became more consistent. 

Previously, different variable names were used: `x_top`, `x_left`, and `top_x`, all for the same coordinate point. With `FrameCoords`, the naming is consistent everywhere.